app/src/main/java/bg/zahov/app/ServiceErrorStateViewModel.kt

```diff
@@ -1,36 +1,50 @@
 package bg.zahov.app
 
-import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import bg.zahov.app.data.interfaces.ServiceErrorHandler
 import bg.zahov.app.data.model.ServiceState
-import bg.zahov.fitness.app.R
+import bg.zahov.app.data.model.state.ShutDownData
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-class ServiceErrorStateViewModel(application: Application) : AndroidViewModel(application) {
-    private val serviceErrorHandler by lazy {
-        application.getServiceErrorProvider()
-    }
-    private val _serviceState = MutableLiveData<State>()
-    val serviceState: LiveData<State>
-        get() = _serviceState
+/**
+ * ViewModel that manages the service error state and updates the UI based on the service status.
+ * It listens to changes in the service state and updates the internal state accordingly.
+ *
+ * @property serviceErrorHandler The ServiceErrorHandler used to observe service state changes.
+ */
+class ServiceErrorStateViewModel(
+    private val serviceErrorHandler: ServiceErrorHandler = Inject.serviceErrorHandler
+) : ViewModel() {
+
+    //The state contains flags for whether the service is shutting down or needs to navigate to shutting down screen.
+    private val _serviceState = MutableStateFlow(ShutDownData())
+
+    /**
+     * Exposes the current shutdown state as a [StateFlow].
+     * Observers can collect this to react to state changes.
+     */
+    val serviceState: StateFlow<ShutDownData> = _serviceState
 
+    /**
+     * Initializes the ViewModel and starts observing the service state.
+     * Based on the observed service state, the internal state is updated.
+     */
     init {
         viewModelScope.launch {
-            serviceErrorHandler.observeServiceState().collect {
-                when (it) {
-                    ServiceState.Unavailable -> _serviceState.postValue(State.NavigateToTimer(action = R.id.to_shutting_down_fragment))
-                    ServiceState.Shutdown -> _serviceState.postValue(State.Shutdown(true))
+            serviceErrorHandler.observeServiceState().collect { serviceState ->
+                when (serviceState) {
+                    ServiceState.Unavailable -> _serviceState.update {
+                        it.copy(navigateToShuttingDown = true)
+                    }
+
+                    ServiceState.Shutdown -> _serviceState.update { it.copy(shutDown = true) }
                     else -> {}
                 }
             }
         }
     }
```

Comment:
This block can be removed. You can simplify the code by using the same `ServiceState` in the view layer, or by mapping the flow to the domain-layer model `ShutDownData()` when declaring the `serviceState` flow.
e.g. 

```
val serviceState: StateFlow<ShutDownData> = serviceErrorHandler.observeServiceState()
.map {  } // your transformation
.launchIn(viewModelScope)
```

Reply:
I will just observe serviceError in main activity